synapse/rest/media/v1/preview_url_resource.py

```diff
@@ -213,14 +220,17 @@ def _do_preview(self, url, user, ts):
         elif _is_html(media_info['media_type']):
             # TODO: somehow stop a big HTML tree from exploding synapse's RAM
 
-            file = open(media_info['filename'])
-            body = file.read()
-            file.close()
+            with open(media_info['filename'], 'rb') as file:
+                body = file.read()
 
             # clobber the encoding from the content-type, or default to utf-8
             # XXX: this overrides any <meta/> or XML charset headers in the body
             # which may pose problems, but so far seems to work okay.
-            match = re.match(r'.*; *charset=(.*?)(;|$)', media_info['media_type'], re.I)
+            match = re.match(
+                r'.*; *charset="?(.*?)"?(;|$)',
+                media_info['media_type'],
+                re.I
+            )
```

Comment:
Is this just a style change?

Reply:
No - the bugfix is the `"?`s on either side which will strip quotes from `charset="utf-8"`, which used to break us. I had to reformat it as the line got too long.